docs/how_to_guides/how_to_use_ui_api.rst

```diff
@@ -0,0 +1,288 @@
+For details see :func:`export_variables`.
+
+For example, if you wanted to mark "flow_vol" as read-only (loading it into the model will not change it), then you could do::
+
+    export_variables(self, name="Feed Z0", desc="Zero-Order feed block",
+                         variables={"flow_vol": {"readonly": True},
+                                    "conc_mass_comp": {}})
+
+
+.. image:: /_static/terminal-icon.png
+    :height: 30px
+    :align: left
+
+.. _howto_api-ui_create-interface:
+
+Creating an interface to a flowsheet
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are three steps for **creating a user interface** to a flowsheet:
+
+1. :ref:`Define a function called flowsheet_interface() <howto_api-ui_define-interface>` that creates the :class:`FlowsheetInterface` object.
+2. Define flowsheet actions. See :ref:`howto_api-ui_create-action`.
+   All interfaces should define the following actions:
+
+    build
+        Build a flowsheet. At the end this will set the built flowsheet into the provided 'ui' parameter with ``ui.set_block(<flowsheet-that-was-built>)``.
+
+    solve
+        Solve a flowsheet
+
+3. :ref:`Set the action functions into the flowsheet interface <howto_api-set-action>` before returning it to the user.
+
+
+.. image:: /_static/terminal-icon.png
+    :height: 30px
+    :align: left
+
+.. _howto_api-ui_define-interface:
+
+Define the ``flowsheet_interface`` function
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The flowsheet interface is created by a function with this signature:
+
+.. function:: flowsheet_interface() -> FlowsheetInterface
+
+   In other words, it takes no arguments and returns a :class:`FlowsheetInterface` object.
+   This object is not yet connected to an IDAES flowsheet block.
+   The function should return a flowsheet interface, see :ref:`howto_api-ui_create-interface`.
+
+Note that you only need to add variables that are not already exported by the model, and that there are reasonable defaults for things like the name, display_name (same as name), and description. So in most cases this will be a very simple call; the extended version was shown for didactic purposes.
+
+.. image:: /_static/terminal-icon.png
+    :height: 30px
+    :align: left
+
+.. _howto_api-ui_create-action:
+
+Create an "action" function
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The action functions should operate on optional keywords for the flowsheet block and the FlowsheetInterface instance.
+It will call the WaterTAP code to perform the appropriate actions.
+The function should have the following signature:
+
+.. function:: action_function([block=None, ui=None], **kwargs)
+
+    Perform an action.
+
+    :param Block block: Flowsheet block
+    :param FlowsheetInterface ui: FlowsheetInterface instance
+    :param dict kwargs: Additional key/value pairs specific to this action
+
+For example::
+
+    def build_flowsheet(ui=None, **kwargs):
```

Comment:
This example snippet may need some additional explanation. 

Where did `my_model` come in from (or is this just meant to be a more generic representation of any model actually being built in this function)? 

It also did not become clear to me how this `build_flowsheet` function is actually used until the last instruction. Maybe consider adding a link to the next instruction saying that this function will be used within your `flowsheet_interface` function.

Reply:
Right, I see how this is confusing. `my_model` was supposed to represent a module (like the metab module) that has functions to build and run the full treatment train. I modified the code to explicitly add an import, and renamed it `my_flowsheet`:

```
    # import module for the full flowsheet
    from watertap.examples.flowsheets.case_studies.foo.bar import my_flowsheet
```

I also added the link, I agree that helps.